conan/tools/system/pip_manager.py

```diff
@@ -9,12 +9,22 @@
 
 class PipEnv:
 
-    def __init__(self, conanfile, folder=None):
+    def __init__(self, conanfile, folder=None, name=""):
+        """
+        :param conanfile: The current conanfile "self"
+        :param folder: Optional folder, by default the "build_folder"
+        :param name: Optional name for the virtualenv, by default "conan_pipenv"
+        """
         self._conanfile = conanfile
-        self.env_name = f"pip_venv_{self._conanfile.name}"
-        self._env_dir = os.path.abspath(os.path.join(folder or self._conanfile.build_folder, self.env_name))
-        self.bin_dir = os.path.join(self._env_dir, "Scripts" if platform.system() == "Windows" else "bin")
-        self._python_exe = os.path.join(self.bin_dir, "python.exe" if platform.system() == "Windows" else "python")
+        name = f"_{name}" if name else ""
+        self.env_name = f"conan_pipenv{name}"
+        self._env_dir = os.path.abspath(os.path.join(folder or conanfile.build_folder,
+                                                     self.env_name))
+        bins = "Scripts" if platform.system() == "Windows" else "bin"
+        self.bin_dir = os.path.join(self._env_dir, bins)
+        pyexe = "python.exe" if platform.system() == "Windows" else "python"
+        self._python_exe = os.path.join(self.bin_dir, pyexe)
+        self._create_venv()
```

Comment:
calling this here is the only functional change / difference.

The rest are styling changes to comply with pep8